sceptre/plan/actions.py

```diff
@@ -1155,7 +1155,7 @@ def dump_config(self):
         """
         return self.stack.config
 
-    @add_stack_hooks
+    @add_stack_hooks_with_aliases([generate.__name__])
```

Comment:
* Running `generate` will invoke the hooks for both `generate` and `dump_template` because the `generate` method invokes `dump_template` and...
* Running `dump_template` will invoke the hooks for both `generate` and `dump_template`